nyt-top-20-science-news/src/components/App/App.jsx

```diff
@@ -0,0 +1,67 @@
+import React, { useEffect, useState } from 'react';
+import Container from 'react-bootstrap/Container';
+import Footer from '../Footer';
+import Header from '../Header';
+import NewsFeed from '../NewsFeed';
+import SearchBox from '../SearchBox';
+
+/**
+ * Main application
+ */
+function App() {
+  const [news, setNews] = useState([]);
+  const [filteredNews, setFilteredNews] = useState(null);
+
+  const TOP_NEWS_ENDPOINT = 'https://api.nytimes.com/svc/topstories/v2/science.json?api-key=';
+  const ARTICLE_COUNT_LIMIT = 20;
+
+  /**
+   * Retrieve NYT Top Science Stories
+   */
+  async function retrievePopularStories() {
+    try {
+      const response = await fetch(TOP_NEWS_ENDPOINT + process.env.REACT_APP_NYT_API_KEY);
+      const { results } = await response.json();
+
+      /**
+       * Sort the news in a descending order according to the update date and time.
+       * In other words, fresh news will be seen first. Additionally, we only
+       * want the first 20 stories, so we chop off the rest. Please consult README.md
+       * to learn more about this decision.
+       */
+      const TOP_STORIES = results
+        ?.sort((a, b) => new Date(b.updated_date) - new Date(a.updated_date))
+        ?.slice(0, ARTICLE_COUNT_LIMIT);
+
+      setNews(TOP_STORIES);
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error('Error:', error);
+    }
+  }
+
+  /**
+   * The empty dependency array (second arg => []) makes sure this effect
+   * runs only once. Similar to componentDidMount.
+   */
+  useEffect(() => {
+    retrievePopularStories();
+  }, []);
+
+  return (
+    <>
```

Comment:
What does the empty bracket allow you to do here? does it end up generating any markup? a `div`?

Reply:
It's [the short syntax for fragments](https://reactjs.org/docs/fragments.html). [Fragments](https://reactjs.org/docs/fragments.html) let you group a list of children without adding extra nodes to the DOM.